lib/iris/fileformats/netcdf.py

```diff
@@ -1339,6 +1339,27 @@ def _get_dim_names(self, cube):
                 dimension_names.append(dim_name)
         return dimension_names
 
+    @staticmethod
+    def cf_valid_var_name(var_name):
+        """
+        Return a valid CF var_name given a potentially invalid name.
+
+        Args:
+
+        * var_name (str):
+            The var_name to normalise
+
+        Returns:
+            A var_name suitable for passing through for variable creation.
+
+        """
+        # Replace invalid charaters with an underscore ("_").
+        var_name = re.sub(r'[^a-zA-Z0-9]', "_", var_name)
+        # Ensure the variable name starts with a letter.
+        if re.match(r'^[^a-zA-Z]', var_name):
+            var_name = 'var_{}'.format(var_name)
+        return var_name
```

Comment:
According to [this by unidata](https://www.unidata.ucar.edu/support/help/MailArchives/netcdf/msg10684.html), it is valid to have multibyte, UTF-8 encoded, NFC-normalized Unicode characters as part of the naming convention for netCDF variables.

I guess we should also support that, otherwise we may impact some users. 

I had a peek at the CF Metedata conventions for [variables](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/cf-conventions.html#variables), but it appears to state that the convention doesn't standardise variable names.

Reply:
Do we need to also consider `dimension` and `attribute` names?

Reply:
> Do we need to also consider dimension and attribute names?

Quite possibly, yes. I kept this one limited, but happy to creep the scope if you force me 😄